crucible/src/Lang/Crucible/Backend.hs

```diff
@@ -311,6 +313,9 @@ class (IsSymInterface sym, HasSymInterface sym bak) => IsSymBackend sym bak | ba
   resetAssumptionState :: bak -> IO ()
   resetAssumptionState bak = restoreAssumptionState bak PG.emptyGoalCollector
 
+  -- | Render a human-readable representation of the state of the backend
+  debugPrintBackendState :: bak -> IO (PP.Doc ann)
```

Comment:
This method feels a bit off to me, considering that it combines two different things (retrieving the backend state and printing it) into a single method. I would be more inclined to add a `getAssumptionState :: bak -> IO (AssumptionState sym)` method that retrieves the current `AssumptionState`, but without doing any of the snapshot-related stuff that `saveAssumptionState` does. Then you could pretty-print the retrieved state using `ppAssumptionState`.

Reply:
Agreed, done.